src/zenml/integrations/vllm/services/vllm_deployment.py

```diff
@@ -0,0 +1,131 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the vLLM Inference Server Service."""
+
+from typing import Any, Optional
+
+from zenml.logger import get_logger
+from zenml.services import (
+    LocalDaemonService,
+    LocalDaemonServiceConfig,
+    ServiceType,
+)
+from zenml.services.service import BaseDeploymentService
+
+logger = get_logger(__name__)
+
+
+class VLLMServiceConfig(LocalDaemonServiceConfig):
+    """vLLM service configurations."""
+
+    blocking: bool = True
+    model: Optional[str] = None
+    tokenizer: Optional[str] = None
```

Comment:
We need to add all other config we want here, but maybe let's start only with essential one as start
tokenizer_mode, trust_remote_code, dtype, revision, served_model_name,

Reply:
Added it in.